de-lib/src/main/java/org/iplantc/de/pipelines/client/presenter/PipelineViewPresenter.java

```diff
@@ -450,15 +464,12 @@ public void onRemoveAppClicked() {
 
         if (selectedApp != null) {
             ListStore<PipelineTask> store = view.getPipelineAppStore();
-
-            store.remove(selectedApp);
-
-            for (int step = 1; step <= store.size(); step++) {
+            boolean removed = store.remove(selectedApp);
+            for (int step = store.size(); step > 0; step--) {
                 PipelineTask app = store.get(step - 1);
-                app.setStep(step);
+                app.setStep(step - 1);
```

Comment:
Ahh! I left debugging code around. Will remove it. As far as `step` number is concerned, it still starts from `1`. See `AppColumnModel.java`->` return object.getStep() + 1;`